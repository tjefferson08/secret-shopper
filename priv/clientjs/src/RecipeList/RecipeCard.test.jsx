import React from 'react';
import RecipeCardBs from './RecipeCard.bs';
import configureMockStore from 'redux-mock-store';
import { render, cleanup } from 'react-testing-library';

const RecipeCard = RecipeCardBs.jsComponent;

let store;

afterEach(cleanup);

const setupTest = () => ({
  props: {
    recipe: {
      name: 'Beans and rice',
      cook_time: '5 minutes',
      prep_time: '10 minutes',
      total_time: '20 minutes',
      ingredients: [
        { id: 1, name: '1 can black beans' },
        { id: 2, name: '1 cup basmati rice' }
      ],
      instructions: [
        { id: 'abc', text: 'microwave ingredients' },
        { id: 'def', text: 'eat food' }
      ]
    },
    showDetails: false
  },
  store: configureMockStore([])()
});

test('it should render the main recipe fields', () => {
  const { props, store } = setupTest();
  const { container } = render(<RecipeCard store={store} {...props} />);
  expect(container.firstChild).toMatchSnapshot();
});

test('it should render all recipe fields when showDetails is true', () => {
  const { props, store } = setupTest();
  props.showDetails = true;
  const { container } = render(<RecipeCard store={store} {...props} />);
  expect(container.firstChild).toMatchSnapshot();
});
