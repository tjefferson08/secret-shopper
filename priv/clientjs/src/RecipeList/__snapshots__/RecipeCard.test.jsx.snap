// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it should render all recipe fields when showDetails is true 1`] = `
<div
  class="pure-u-1 pure-u-md-1-3"
>
  <div
    class="recipe-card"
  >
    <div
      class="header"
    >
      <i
        class="far fa-star fa-2x"
        data-testid="favorite-badge"
      />
      <h3
        class="name"
      >
        Beans and rice
      </h3>
    </div>
    <img
      class="thumbnail"
      alt="Beans and rice"
    />
    <div
      class="time-container"
    >
      <strong
        class="time-label"
      >
        Cook time:
      </strong>
      <p
        class="time-value"
      >
        5 minutes
      </p>
    </div>
    <div
      class="time-container"
    >
      <strong
        class="time-label"
      >
        Prep time:
      </strong>
      <p
        class="time-value"
      >
        10 minutes
      </p>
    </div>
    <div
      class="time-container"
    >
      <strong
        class="time-label"
      >
        Total time:
      </strong>
      <p
        class="time-value"
      >
        20 minutes
      </p>
    </div>
    <h4>
      Ingredients
    </h4>
    <ul>
      <li>
        1 can black beans
      </li>
      <li>
        1 cup basmati rice
      </li>
    </ul>
    <h4>
      Instructions
    </h4>
    <ol>
      <li>
        microwave ingredients
      </li>
      <li>
        eat food
      </li>
    </ol>
  </div>
</div>
`;

exports[`it should render the main recipe fields 1`] = `
<div
  class="pure-u-1 pure-u-md-1-3"
>
  <div
    class="recipe-card"
  >
    <div
      class="header"
    >
      <i
        class="far fa-star fa-2x"
        data-testid="favorite-badge"
      />
      <h3
        class="name"
      >
        Beans and rice
      </h3>
    </div>
    <img
      class="thumbnail"
      alt="Beans and rice"
    />
    <div
      class="time-container"
    >
      <strong
        class="time-label"
      >
        Cook time:
      </strong>
      <p
        class="time-value"
      >
        5 minutes
      </p>
    </div>
    <div
      class="time-container"
    >
      <strong
        class="time-label"
      >
        Prep time:
      </strong>
      <p
        class="time-value"
      >
        10 minutes
      </p>
    </div>
    <div
      class="time-container"
    >
      <strong
        class="time-label"
      >
        Total time:
      </strong>
      <p
        class="time-value"
      >
        20 minutes
      </p>
    </div>
  </div>
</div>
`;
